File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hashTable = {}
    for(let i=0; i < nums.length; i++){
        const pairIndex = hashTable[nums[i]]
        if(pairIndex === undefined){
            hashTable[target - nums[i]] = i
        }else{
            return [pairIndex, i]
        }
    }
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(len(nums)) :
            pairIndex = dict.get(nums[i])
            if pairIndex != None:
                return [pairIndex, i]
            else :
                dict[target - nums[i]] = i